hat/assets/js/apps/Iaso/domains/home/components/LangSwitch.tsx

```diff
@@ -46,21 +62,31 @@ export const LangSwitch: FunctionComponent = () => {
     );
     return (
         <>
-            {APP_LOCALES.map((locale, index) => (
-                <Box key={locale.code}>
-                    <Box
-                        className={classNames(
-                            classes.languageSwitch,
-                            locale.code === activeLocale &&
-                                classes.languageSwitchActive,
-                        )}
-                        onClick={() => handleClick(locale.code)}
-                    >
-                        {locale.code}
+            {APP_LOCALES.map((locale, index) => {
+                const switchStyle =
+                    topBar && locale.code !== activeLocale
+                        ? classes.languageSwitchOnTopBar
+                        : classes.languageSwitch;
+                const switchActiveStyle =
+                    locale.code === activeLocale &&
+                    (topBar
+                        ? classes.languageSwithActiveOnTopBar
+                        : classes.languageSwitchActive);
+                return (
+                    <Box key={locale.code}>
+                        <Box
+                            className={classNames(
+                                switchStyle,
+                                switchActiveStyle,
+                            )}
+                            onClick={() => handleClick(locale.code)}
+                        >
+                            {locale.code}
+                        </Box>
+                        {index + 1 !== APP_LOCALES.length && '-'}
```

Comment:
there should be a most cleaner way to do that, eventually by splitting each iteration in a separate component or a better use of classnames.
Also i'm not a big fan the underline and the color of the selected language. It should use secondary color without underline if selected, and keep it white if not selected.